src/modules/Mask/Module.js

```diff
@@ -0,0 +1,82 @@
+module.exports = function Mask(options, UI, util) {
+  var defaults = require('./../../util/getDefaults.js')(require('./info.json'));
+  options.offset = options.offset || defaults.offset;
+  options.resize = options.resize || defaults.resize;
+
+  var output;
+
+  // This function is called on every draw.
+  function draw(input, callback, progressObj) {
+    progressObj.stop(true);
+    progressObj.overrideFlag = true;
+
+    var step = this;
+
+    var getPixels = require('get-pixels');
+
+    // convert offset as string to int
+    if (typeof options.offset === 'string')
+      options.offset = parseInt(options.offset);
+
+    // save first image's pixels
+    var priorStep = this.getStep(options.offset);
+
+    if (priorStep.output === undefined) {
+      this.output = input;
+      UI.notify('Offset Unavailable', 'offset-notification');
+      callback();
+    }
+
+    const alpha_masking = function(c, a) {
+      return (a * c + (255 - a) * c) / 255;
+    };
+    const internalSequencer = ImageSequencer({ inBrowser: false, ui: false });
+    internalSequencer.loadImage(priorStep.output.src, function() {
+      internalSequencer.importJSON([{ 'name': 'resize', 'options': { resize: options.resize } }]);
```

Comment:
Can this be done without using an internal sequencer, please?  Internal sequencer has a ton of overhead. Try using the edge-detect approach (check to code to see how it uses Blur Module).

Reply:
I tried that but wasn't able to do that.

Reply:
I don't think that can be done as of now. Refer: https://github.com/publiclab/image-sequencer/issues/1560

**P.S.**: If you use the current approach, it won't work with GIFs. Please add that to the end of the README.

Reply:
Ok, So I have opened a pull request to refactor resize #1562